### ISSUE TITLE
No available UPnP device found

### ISSUE THREAD

**Issue by VeryKnave** [quasar/QuasarRAT#624]
### I made sure:
- firewall is not blocking Quasar;
- TCP and UDP ports are added to exceptions of firewall;
- my router has UPnP enabled;
- _no-IP's provided IP_/_localhost_ was used to create the executable;
- no-IP's DUC is running;

![capture](https://user-images.githubusercontent.com/23298862/31389841-e7c2e0a8-adda-11e7-9189-875a37d10f3c.PNG)

I am using Windows 10 pro 64-bit.

**Comment by VeryKnave** [comment]
I kept the UPnP checked, left automatic listening checked, pressed save and restarted Quasar (in the bottom left corner says: "Listening to port 1177").
Do you meant to check this:
![fiew](https://user-images.githubusercontent.com/23298862/31408016-b38f0520-ae0f-11e7-9e92-5ef284cd55b5.PNG)
? If not, how can I do it? I looked on the internet for about 30 minutes and I couldn't find anything.

**Comment by MaxXor** [comment]
This is the local firewall and has nothing to do with UPnP, it's set up correctly for Quasar (on port 1177). UPnP allows Quasar to automatically forward the port in your router firewall. What's the problem exactly? You get no external connections?

**Comment by VeryKnave** [comment]
I transfered the built executable to my second laptop, ran it, the proccess runs in the background but I get no client on my first laptop.

**Comment by Abdullah2993** [comment]
@VeryKnave almost all routers have a UPnP table you can check if your modem/routers show the entry for quasar on that port?

**Comment by VeryKnave** [comment]
I've made some test.

first laptop      -> Windows 10 64-bit
second laptop -> Windows 7 32-bit

- I tried QuasarRAT on my _second laptop_ (the same port: 1177) and the UPnP message didn't displayed when I started the listening. I sent the build executable from my _second laptop_ to my _first laptop_ and it still didn't worked.
- I tried running the executable from the laptop that it was created on. It worked on both laptops so I can control only "my own laptop".
- It works only if I add **localhost** as a connection host in the builder, but still only on the laptop that is listening. If I have only the no-ip provided host as a connection host, it wouldn't even work on the laptop that is listening.

 @Abdullah2993 How can I check that?

**Comment by MaxXor** [comment]
What IP do you enter in the builder? Your local IP or external IP (visible by http://whatismyip.org/)

**Comment by VeryKnave** [comment]
@MaxXor I enter the domain name provided by no-IP (`name`.zapto.org), so the external IP. Does it make any difference?

**Comment by MaxXor** [comment]
No, do you know how to forward the port in your router?

**Comment by VeryKnave** [comment]
I am not sure. This is what I did:
![capture](https://user-images.githubusercontent.com/23298862/31463048-6d6b0e0e-aed7-11e7-92e5-cc34bf2e202a.PNG)
The internal and external IPs are correct. It still doesn't work.

**Comment by Abdullah2993** [comment]
@VeryKnave leave remote host empty. After that run something like netcat on port `1177` and use http://canyouseeme.org/ to verify the port is open

**Comment by VeryKnave** [comment]
@Abdullah2993 Okay, I left it empty. What command should I use or how should I do it on Windows?

**Comment by Abdullah2993** [comment]
@VeryKnave just download the `netcat` binary

**Comment by VeryKnave** [comment]
I tried this command: `nc -vv -z -w3 <my external ip> <port>`.

![capture](https://user-images.githubusercontent.com/23298862/31548255-f1243b1a-b032-11e7-9a0d-c4d16d404360.PNG)
I listened with Quasar on port 1177 while running the commands.

Also http://canyouseeme.org/ and other online port checkers seem to not identify port 80 and any other ports as open.
![capture2](https://user-images.githubusercontent.com/23298862/31548459-869fd578-b033-11e7-9b28-3e0c5ad2478a.PNG)

I don't understand what I am doing wrong.

**Comment by MaxXor** [comment]
Your router or windows firewall is blocking the port. Or maybe even your ISP doesn't allow open ports.

**Comment by Abdullah2993** [comment]
without running quasar. run `nc -lnvp 1177` and check for port `1177` from canyouseem.org

**Comment by VeryKnave** [comment]
I did it, but it still doesn't work.
![capture](https://user-images.githubusercontent.com/23298862/31611059-37ed334c-b283-11e7-9702-036dc9281293.PNG)

I have tried another 3 or 4 different RATs and none of them were working on my Windows laptops.
I have tried a single RAT on my Linux (dual-boot) and it worked.

Also, I deleted Quasar and downloaded it again and the "No available UPnP device found" message does not pop anymore.
